apps/src/music/views/MusicPlayView.tsx

```diff
@@ -3,25 +3,37 @@ import React, {memo, useCallback, useMemo} from 'react';
 import {Button} from '@cdo/apps/componentLibrary/button';
 import {Heading2, BodyTwoText} from '@cdo/apps/componentLibrary/typography';
 import commonI18n from '@cdo/locale';
-import codeOrgLogo from '@cdo/static/code-dot-org-white-logo.svg';
+import musicPlayViewLogo from '@cdo/static/music/music-play-view.png';
 
 import moduleStyles from './music-play-view.module.scss';
 
 import musicI18n from '../locale';
+import {useAppSelector} from '@cdo/apps/util/reduxHooks';
+import {queryParams} from '@cdo/apps/code-studio/utils';
+import Lab2Registry from '@cdo/apps/lab2/Lab2Registry';
 
 interface MusicPlayViewProps {
-  isPlaying: boolean;
-  onPlay: () => void;
-  onStop: () => void;
-  projectName: string;
+  setPlaying: (value: boolean) => void;
 }
 
 const MusicPlayView: React.FunctionComponent<MusicPlayViewProps> = ({
-  isPlaying,
-  onPlay,
-  onStop,
-  projectName,
+  setPlaying,
 }) => {
+  const {isPlaying, lastMeasure, currentPlayheadPosition} = useAppSelector(
+    state => state.music
+  );
+  const progressValue =
+    lastMeasure === 1
+      ? 0
+      : ((currentPlayheadPosition - 1) / (lastMeasure - 1)) * 100;
+  const progressSliderValue = progressValue.toString();
```

Comment:
Out of curiosity, why is this conversion necessary?

Reply:
From https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/range - The [value](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#value) attribute contains a string which contains a string representation of the selected number. The value is never an empty string ("").

But I believe it works as a number as well. I was also thinking since`min` and `max` are defined as strings, just being consistent. But looks like `min` and `max` could be assigned numbers. I don't have a preference on this. Can change if you prefer as numbers.

Reply:
Gotcha, no strong preference either way, what you have is fine!